sentry/src/main/java/io/sentry/FilterString.java

```diff
@@ -3,21 +3,37 @@
 import java.util.Objects;
 import java.util.regex.Pattern;
 import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
 
 public final class FilterString {
   private final @NotNull String filterString;
-  private final @NotNull Pattern pattern;
+  private final @Nullable Pattern pattern;
 
   public FilterString(@NotNull String filterString) {
     this.filterString = filterString;
-    this.pattern = Pattern.compile(filterString);
+    @Nullable Pattern pattern = null;
+    try {
+      pattern = Pattern.compile(filterString);
+    } catch (Throwable t) {
+      Sentry.getCurrentScopes()
+          .getOptions()
+          .getLogger()
+          .log(
+              SentryLevel.DEBUG,
```

Comment:
maybe worth raising the level to Warning?

Reply:
Could also be intentional that a filter String isn't meant as a regex and just contains chars not applicable for a regex. I wouldn't consider that something worth WARNING about.